dev-aws/kafka-shared-msk/pubsub/tf-kafka-config-admin.tf

```diff
@@ -0,0 +1,28 @@
+# Grant pubsub/tf-kafka-config-admin permission to all the resources
+resource "kafka_acl" "tf_kafka_config_admin_topic" {
+  resource_name       = "*"
+  resource_type       = "Topic"
+  acl_principal       = "User:CN=pubsub/tf-kafka-config-admin"
+  acl_host            = "*"
+  acl_operation       = "All"
+  acl_permission_type = "Allow"
+}
+
+resource "kafka_acl" "tf_kafka_config_admin_group" {
+  resource_name       = "*"
+  resource_type       = "Group"
+  acl_principal       = "User:CN=pubsub/tf-kafka-config-admin"
+  acl_host            = "*"
+  acl_operation       = "All"
+  acl_permission_type = "Allow"
+}
+
+resource "kafka_acl" "tf_kafka_config_admin_cluster" {
+  resource_name                = "kafka-cluster"
+  resource_type                = "Cluster"
+  acl_principal                = "User:CN=pubsub/tf-kafka-config-admin"
+  acl_host                     = "*"
+  acl_operation                = "All"
+  acl_permission_type          = "Allow"
+  resource_pattern_type_filter = "Literal"
```

Comment:
I was wondering why we only set this one on the config setup, but I see the docs say
The docs (https://registry.terraform.io/providers/Mongey/kafka/latest/docs/resources/acl#resource_pattern_type_filter) say:

> [resource_pattern_type_filter](https://registry.terraform.io/providers/Mongey/kafka/latest/docs/resources/acl#resource_pattern_type_filter) - (Optional) The pattern filter. Valid values are Prefixed, Any, Match, Literal. Default Literal.

so I guess it's because we set the `resource_name` here. But for the others won't this default to `Literal` so e.g. we would only have permission on the topic named literally: `"*"`

Reply:
Indeed `Literal` is the default, but you can use it to match too. See https://docs.confluent.io/platform/current/security/authorization/acls/overview.html#use-prefixed-acls